src/transport/smtp/client/async_net.rs

```diff
@@ -464,6 +464,50 @@ impl AsyncNetworkStream {
         }
     }
 
+    pub fn certificate_chain(&self) -> Result<Vec<Vec<u8>>, Error> {
+        match &self.inner {
+            #[cfg(feature = "tokio1")]
+            InnerAsyncNetworkStream::Tokio1Tcp(_) => {
+                Err(error::client("Connection is not encrypted"))
+            }
+            #[cfg(feature = "tokio1-native-tls")]
+            InnerAsyncNetworkStream::Tokio1NativeTls(_) => panic!("Unsupported"),
```

Comment:
Same here for https://docs.rs/native-tls/0.2.12/native_tls/struct.TlsStream.html#method.peer_certificate

Reply:
See my comment above. There is https://github.com/sfackler/rust-native-tls/pull/117 which implements `certificate_chain` for native-tls, but seems abandoned.